SignalOutput

simonmeaden opened the pull request:
Hi, I've been using your logger for a while but needed to output via a Qt signal/slot connection. To do this I had to modify the original project, adding a SignalOutput class. this made it necessary to make SignalOutput, Configuration and QLogger extend QObject i order to access the Signal capability, then routethe signal automatically from SignalOutput via Configuration to QLogger::logMessage(QString) as there is no direct access to the SignalOutput class. I also Split a separate BaseOutput class to allow me to use formatLogText() without the QTextStream in ConsoleObject which is not needed for SignalOutput.

Configuration and ConfigFileHandler have been modified to allow the output type to be set in qlogger.cfg. However I noticed while testing that the setup as stands before I modified it, doesn't actually use the qlogger.cfg file as recursiveFolderSearch() only searches for directories, no files were actually searched so I changed the filter from QDir::Dirs|Dir::NoDotAndDotDot to QDir::Dirs|QDir::Files|QDir::NoDotAndDotDot.

The README file has been amended to show the changes.

All a user has to do is select an OutputType of SIGNAL and make a connection to his recipient class.

Commit:
SignalOutput class added.
This allows messages to be output through the QT signal slot
mechanism.

Commit:
qlogger.pro added (alternative to QLogger.pro). This should be ignored
as it it is only used to handle the rather idiosyncratic build method I
normally use.

Commit:
Tested both as an standalone QLogger::addLogger() and a config file
logger addition.

Commit:
Modified qlogger.pro to build both static and shared libraries.

sfadiga commented:
Hi Simon,
Thanks for your interest in qlogger ! To be honest its been quite a while since I touched the source code !

I like the idea behind what you've done, although I couldn't get it working here at my dev environment, I've spot one small bug on the configurationhandler at the recursiveFolderSearch method, if this method does not find the qlogger.cfg file it will loop indefinitely on the recursion, but no biggie this is simple stuff to fix. Can you provide me your test project so I can see the logger working by emitting signals ?

Regards,
Sandro.

simonmeaden commented:
Yes no problem. I will look at that bug first and maybe play around with it some more to test it is working properly.

Regars
Simon

Commit:
Daily update

Commit:
Optional console/gui support added to main.cpp.
MainWindow + ui added.

Commit:
Change log level capability added.

Commit:
Change log level capability added. Not tested at this point.

Commit:
Completed testing. QLogger modified for SIGNAL operation. QLogger
modified to allow the level to be modified. Testing GUI working.

simonmeaden commented:
OK I think I fixed that bug. Also added a gui testing system and added a method to allow the level to be modified for a particular owner/type combo. Let me know if you have any problems with it.

sfadiga commented:
Hi Simon, I'm looking at the code and I'm making some changes (big ones :)
there were some leaks and some abstraction issues that I'm revisiting, I'll
commit that change soon, but if you wan't to keep changing the current
version I can branch it, or you can fork it on your own Repo, as soon as I
have somthing working here I'll let you know.

Regards
Sandro.


On Tue, Sep 18, 2018 at 8:57 AM Simon Meaden <notifications@github.com>
wrote:

> OK I think I fixed that bug. Also added a gui testing system and added a
> method to allow the level to be modified for a particular owner/type combo.
> Let me know if you have any problems with it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sfadiga/QLogger/pull/1#issuecomment-422363980>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AB7ZqGI6LEAHQZW_iKH602LpgfdqGEZIks5ucN-pgaJpZM4Wf3oL>
> .
>

simonmeaden commented:
Well it's working for what I need it for at the moment so I'll hang on and see your changes and then merge my stuff with the new code.

sfadiga commented:
Hi Simon, its been a while, but I've commited a big change in the project (since there was so much change I added also the signal/slot option on my version I hope you don't mind). I suggest you to take a look at this new 'version' since there is so much improvements than the original one. The way to use the logger stills the same (with some minor changes). Regards, Sandro.

sfadiga commented:
I will close this pull request since it will not work with the changes I've done.

simonmeaden commented:
Sorry for not replying earlier but my computer crashed and burned and i' e been away and only just fixed it. No problem with you incorporating the signal into the main branch. I'll check out the new branch when I've time and see it does what I need.
regards
Simon
